File: business-easy-core/src/main/java/com/businesseasy/core/entities/UnitConversionEntity.java
package com.businesseasy.core.entities;

import com.businesseasy.core.common.enums.Operator;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "unit_conversion")
public class UnitConversionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    Long id;

    @Column(name = "fromUnit")
    Long from;

    @Column(name = "toUnit")
    Long to;

    @Column(name = "cal_step")
    Integer calStep;

    @Enumerated(EnumType.STRING)
    @Column(name = "operator")
    Operator operator;

    @Column(name = "constant")
    BigDecimal constant;
}

File: easiness-ui/app/server/entity/unit-conversion.entity.ts
import Big from "big.js"
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm"
import { Operator } from "../model/operator"

@Entity({ name: 'unit_conversion' })
export class UnitConversion {
    @PrimaryGeneratedColumn()
    id: number

    @Column({ name: 'from_unit' })
    from: number

    @Column({ name: 'to_unit' })
    to: number

    @Column({ name: 'cal_step' })
    calStep: number

    @Column()
    operator: Operator

    @Column('numeric', {
        precision: 20,
        scale: 6,
        transformer: {
            from: (value: string) => new Big(value),
            to: (value: Big) => value.toString()
        }
    })
    constant: Big;
}